agave_app/GLView3D.cpp

```diff
@@ -309,14 +309,14 @@ GLView3D::toggleAreaLightRotateControls()
 {
   // toggle rotate tool
   if (m_areaLightMode == AREALIGHT_MODE::NONE || m_areaLightMode == AREALIGHT_MODE::TRANS) {
-    m_viewerWindow->showAreaLightGizmo(true);
+    m_viewerWindow->select(m_viewerWindow->sceneView.scene->SceneAreaLight());

```

Comment:
key change: now we just tell the viewerWindow what the selected SceneObject is.  If the object provides a Tool for drawing itself when selected (e.g. a wireframe box), then that will automatically be added.